File: downloader/core/src/commonMain/kotlin/com/flowspeed/lib/downloader/downloaditem/http/IHttpBasedDownloadCredentials.kt
package com.flowspeed.lib.downloader.downloaditem.http

import com.flowspeed.lib.downloader.downloaditem.IDownloadCredentials

interface IHttpBasedDownloadCredentials : IDownloadCredentials {
    val headers: Map<String, String>?
    val username: String?
    val password: String?
    val userAgent: String?
}

File: FlowDownloadManager/src/Flow.Core/Models/IHttpBasedDownloadCredentials.cs
using System.Collections.Generic;

namespace Flow.Core.Models;

public interface IHttpBasedDownloadCredentials : IDownloadCredentials
{
    Dictionary<string, string>? Headers { get; set; }
    string? Username { get; set; }
    string? Password { get; set; }
    string? UserAgent { get; set; }
}
